prboom2/src/dsda/console.c

```diff
@@ -254,6 +260,13 @@ static void dsda_ExecuteConsole(void) {
         break;
       }
     }
+
+    char prefix[2];
+    memcpy(prefix, command, 2);
+    if(!stricmp(prefix, "id"))
```

Comment:
There shouldn't be any exceptions in the console, so we don't want to have an extra specific comparison here. It adds extra complexity that is avoidable. Rather we should follow the existing behaviour. I'd suggest adding a new command "cheat.enter" with the arg for the cheat. Define it so that it automatically covers all current and future cheat codes.

I should be able to enter for example "cheat.enter idclev23" or "cheat.enter tntem"

Reply:
You can follow the pattern of console_CommandLock (with console_CheatEnter).

Reply:
Typing cheats normally makes me press some keys that move the player, rewind or even restart the level.
I know, that I can bind a certain key to a certain cheat, but sometimes I want to use a cheat that I have no keybind for.
The console seemed like the best solution for this problem.

Reply:
Hm, one alternative would be a menu that shows all the cheats and their states active / inactive and you can toggle them. Or there could be a specific key to prompt the user to enter a cheat (that blocks other responders from seeing the inputs). Would either of those be useful for you? Adding the stuff to the console like this is something I wanted to do anyway in the long run, so it's still worth merging, but I may consider one of those things for the future.

Reply:
A key to prompt the user to input a cheat would be perfect.